runtime/Go/antlr/utils.go

```diff
@@ -353,36 +353,37 @@ func PrintArrayJavaStyle(sa []string) string {
 	return buffer.String()
 }
 
-
 // murmur hash
 const (
-	c1_32 = 0xCC9E2D51
-	c2_32 = 0x1B873593
-	n1_32 = 0xE6546B64
+	c1_32 uint = 0xCC9E2D51
+	c2_32 uint = 0x1B873593
+	n1_32 uint = 0xE6546B64
 )
 
 func murmurInit(seed int) int {
 	return seed
 }
 
 func murmurUpdate(h1 int, k1 int) int {
-	k1 *= c1_32
-	k1 = (k1 << 15) | (k1 >> 17) // rotl32(k1, 15)
-	k1 *= c2_32
+	var k1u uint
+	k1u = uint(k1) * c1_32
+	k1u = (k1u << 15) | (k1u >> 17) // rotl32(k1u, 15)
+	k1u *= c2_32
 
-	h1 ^= k1
-	h1 = (h1 << 13) | (h1 >> 19) // rotl32(h1, 13)
-	h1 = h1*5 + 0xe6546b64
-	return h1
+	var h1u = uint(h1) ^ k1u
+	h1u = (h1u << 13) | (h1u >> 19) // rotl32(h1u, 13)
+	h1u = h1u*5 + 0xe6546b64
+	return int(h1u)
 }
 
 func murmurFinish(h1 int, numberOfWords int) int {
```

Comment:
Probably should fix the input type instead of relying on a cast (enforcing that there are no negative numbers passed to this function).  Also might be good to document that numberOfWords is limited to be 4x less than the normal uint range.